contracts/LockedFund/LockedFund.sol

```diff
@@ -623,12 +872,71 @@ contract LockedFund is ILockedFund {
 	}
 
 	/**
-	 * @notice Function to read the cliff and duration of a user.
-	 * @param _addr The address whose cliff and duration has to be found.
-	 * @return The cliff of the user vesting/lock.
-	 * @return The duration of the user vesting/lock.
+	 * @notice Function to read the cliff, duration and Type of a Vesting.
+	 * @param _vestingData The address whose cliff and duration has to be found.
+	 * @return _cliff The cliff of the user vesting.
+	 * @return _duration The duration of the user vesting.
+	 * @return _vestingType The Vesting Type of the user vesting.
+	 */
+	function getCliffDurationAndType(bytes32 _vestingData)
+		external
+		view
+		returns (
+			uint256 _cliff,
+			uint256 _duration,
+			uint256 vestingType
+		)
+	{
+		return (vestingDatas[_vestingData].cliff, vestingDatas[_vestingData].duration, vestingDatas[_vestingData].vestingType);
+	}
+
+	/**
+	 * @notice Returns the Vesting Contract Address.
+	 * @param _tokenOwner The owner of the vesting contract.
+	 * @param _cliff The cliff of the user vesting.
+	 * @param _duration The duration of the user vesting.
+	 * @param _vestingCreationType The Vesting Type of the user vesting.
+	 * @return _vestingAddress The Vesting Contract Address.
+	 */
```

Comment:
it is not obvious that passing address(0) should return vestings of msg.sender and most probably will never be used this way
consider removing it

Reply:
Not sure where it is mentioned that passing address(0) will return the vesting details of `msg.sender`. It takes `_tokenOwner` address to determine the vesting details.